### PR TITLE
feat: improve download progress bar

### LINKED ISSUE #16507: cli: show package size in download progress bar
In compact mode, we download node packages, which mandates us to show the size in the progress bar to keep users informed that it takes time to download the heaviest objects in the universe.

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref https://github.com/denoland/deno/pull/16975

Closes #16507

### CODE REVIEW

File: cli/util/progress_bar.rs [hunk 1]
```diff
@@ -6,6 +6,299 @@ use indexmap::IndexSet;
...
+              *progress_bar.pos.read(),
+              progress_bar.size,
+              progress_bar.style,
+              progress_bar.start_time.elapsed().unwrap(),
+            ));
+
+            eprint!("{}", text);
+          }
+
+          std::thread::sleep(Duration::from_millis(100));
```
Comment (bartlomieju) [inline cli/util/progress_bar.rs:167]: Make it 16 for that sweet 60fps?

**Comment by bartlomieju** [comment]
Tested locally, works way better than current implementation!

**Comment by dsherret** [comment]
Example videos:

https://user-images.githubusercontent.com/1609021/207158847-0cba239e-b4d0-4a58-b94a-1cd5b3b2c0d5.mp4

https://user-images.githubusercontent.com/1609021/207158860-c5432aa4-a0b8-41c4-b909-2b7ba55dff14.mp4

File: cli/util/console.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  deno_runtime::ops::tty::console_size(stderr).ok()
+}
+
+pub fn show_cursor() {
+  eprint!("\x1B[?25h");
+}
+
+pub fn hide_cursor() {
+  eprint!("\x1B[?25l");
+}
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply cli/util/console.rs:15]: Maybe, let's not hide cursor at all? I hit an error (or Ctrl+C, don't remember) while testing this and then my cursor was gone for good, until I restarted my terminal session. Seems iffy...

[the remark this answers is not in the document]
Comment (dsherret) [reply cli/util/console.rs:15]: Oh, that's not good

[the remark this answers is not in the document]
Comment (dsherret) [reply cli/util/console.rs:15]: Ok, the flashing cursor is not that bad. I'll remove this.

File: cli/util/progress_bar.rs (hunk 1 above)
Comment (dsherret) [reply cli/util/progress_bar.rs]: Slow computers might struggle. I think we might have to add some more code that times how long a draw takes then update the fps based on that if we want to go down to 16.

**Comment by bartlomieju** [comment]
LGTM, but I can't approve @dsherret since it's my PR

**Review by dsherret — APPROVED** [review]
LGTM. I'll merge it in a few seconds

**Pull request merged**